CONTRIBUTING.md

```diff
@@ -0,0 +1,131 @@
+
+
+## Issues
+If you encounter any [issues](https://github.com/pinecone-io/canopy/issues/new/choose) while using the project, please report them. 
+Include a detailed description of the problem, steps to reproduce it, and your environment details.
+
+For any question, please use the `Discussions` section rather than opening an issue. This helps keep the issue tracker 
+focused on bugs and feature requests.
+
+
+## Pull Requests
+
+It is really simple to get started and create a pull request. Canopy is released regularly, so, you should see your
+improvements released in a matter of days or weeks 🚀
+
+Unless your change is trivial (typo, docs tweak etc.), please create an issue to discuss the change before
+creating a pull request.
+
+If you're looking for something to get your teeth into, check out the
+["good first issue"](https://github.com/pinecone-io/canopy/issues?q=is:issue+is:open+label:%22good+first+issue%22)
+label on GitHub.
+
+
+### Prerequisites
+
+You'll need the following prerequisites:
+
+- Any Python version between **Python 3.9 and 3.11**
+- [**Poetry**](https://github.com/python-poetry/poetry)
+- [**git**](https://git-scm.com/)
+- [**make**](https://www.gnu.org/software/make/)
+
+### Installation & Setup
+
+Fork the repository on GitHub and clone your fork locally.
+
+```bash
+# Clone your fork and cd into the repo directory
+git clone git@github.com:<your username>/canopy.git
+cd canopy
+
+# Install canopy, dependencies and dev dependencies
+make install
+```
+
+### Check out a new branch and make your changes
+
+Create a new branch for your changes.
+
+```bash
+# Checkout a new branch and make your changes
+git checkout -b my-new-feature-branch
+# Make your changes...
+```
+
+### Document your changes
+When contributing to Canopy, please make sure that all code is well documented. 
+
+The following should be documented using properly formatted docstrings:
+
+- Modules
+- Class definitions
+- Function definitions
+- Module-level variables
+
+Canopy uses [Google-style docstrings](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings) formatted 
+according to [PEP 257](https://www.python.org/dev/peps/pep-0257/) guidelines. 
+(See [Example Google Style Python Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) 
+for further examples.)
+
+[pydoclint](https://github.com/jsh9/pydoclint) is used for linting docstrings. You can run `make lint` to check your docstrings.
+
+If you are making changes to the public API, please update the documentation in the README.md file.
+
+### Add the relevant tests
+After making your changes, make sure to add the relevant tests. If you are adding a new feature, add unit tests.
+If you are fixing a bug, add a regression test. Please make an effort not to copy and paste other tests, but instead
+try to use existing fixtures or create new ones. Also, prefer parametrizing the tests over duplication where possible.
```

Comment:
```suggestion
All code changes to Canopy need to be covered by tests. After making your changes, make sure to add the relevant tests.   
If you are adding a new feature or functionality - please add a unit test by creating a new `test_<module_name>.py` file under the `tests/unit/` directory. Tests that require external integration to (e.g. communication with an API or service) should be placed under the `tests/system/` directory. If you are fixing a bug, add a regression test case to the existing test file.   

Please make an effort to avoid code duplication. Some unit tests have a common base class that can be extended. Other tests use fixtures to parameterize the same test cases over several subclasses. Instead of copy-pasting other test cases, try to utilize these mechanism as much as possible.```